### PR TITLE
Comparisons

### PR DESCRIPTION
**Pull request by rjitsu**
I have translated the comparisons article. Let me know if it needs any correction!

### CODE REVIEW

File: 1-js/02-first-steps/09-comparison/1-comparison-questions/solution.md [hunk 1]
```diff
@@ -9,13 +8,12 @@ undefined === null → false
...
-3. Again, dictionary comparison, first char of `"2"` is greater than the first char of `"1"`.
-4. Values `null` and `undefined` equal each other only.
-5. Strict equality is strict. Different types from both sides lead to false.
-6. Similar to `(4)`, `null` only equals `undefined`.
-7. Strict equality of different types.
+1. जाहिर है, सच है।
+2. शब्दकोश की तुलना, इसलिए झूठी है। `"a"` `"p"` से छोटा है।
+3. फिर से, शब्दकोश तुलना, `"2"` का पहला करैक्टर `"1"` के पहले करैक्टर से अधिक है।
+4. मान `null` और `undefined` एक-दूसरे के बराबर हैं।
+5. सख्त समानता सख्त है। दोनों तरफ से अलग-अलग प्रकार झूठे होते हैं।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/1-comparison-questions/solution.md:17]: This sentence can be improved. The second sentence in particular needs some changes. A good test is to come back to the article after 30 minutes and see if every line is easy to understand. Translation doesn't need to be exact but it should convey the topic easily.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 2]
```diff
@@ -1,214 +1,191 @@
-# Comparisons
+# तुलना
 
-We know many comparison operators from maths.
+हम गणित से कई तुलना ऑपरेटरों को जानते हैं।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:3]: **Operators** here refers to the **notations**, which should be translated as **प्रतीकों**. [Refer to this.](https://hi.wikipedia.org/wiki/%E0%A4%97%E0%A4%A3%E0%A4%BF%E0%A4%A4%E0%A5%80%E0%A4%AF_%E0%A4%AA%E0%A5%8D%E0%A4%B0%E0%A4%A4%E0%A5%80%E0%A4%95%E0%A5%8B%E0%A4%82_%E0%A4%95%E0%A5%80_%E0%A4%B8%E0%A4%BE%E0%A4%B0%E0%A4%A3%E0%A5%80)

File: 1-js/02-first-steps/09-comparison/article.md [hunk 3]
```diff
@@ -1,214 +1,191 @@
...
 
-In JavaScript they are written like this:
+जावास्क्रिप्ट में वे इस तरह लिखे जाते हैं:
 
-- Greater/less than: <code>a &gt; b</code>, <code>a &lt; b</code>.
-- Greater/less than or equals: <code>a &gt;= b</code>, <code>a &lt;= b</code>.
-- Equals: `a == b`, please note the double equality sign `==` means the equality test, while a single one `a = b` means an assignment.
-- Not equals. In maths the notation is <code>&ne;</code>, but in JavaScript it's written as <code>a != b</code>.
+- ग्रेटर / से कम: <code>a &gt; b</code>, <code>a &lt; b</code>।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:7]: We can refer to the link above for some of these notations. Whichever notations are not available (in Hindi) in the list, you can translate them by using an online translator, Use words and provide some context to get more accurate results.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 4]
```diff
@@ -1,214 +1,191 @@
...
+- ग्रेटर / से कम या बराबर: <code>a &gt;= b</code>, <code>a &lt;= b</code>।
+- समतुल्य: `a == b`, कृपया दोहरे समानता चिन्ह पर ध्यान दें `==` का अर्थ है समानता परीक्षण, जबकि एकल `a = b` का अर्थ है असाइनमेंट।
+- नहीं के बराबर है। गणित में अंकन<code>&ne;</code>है, लेकिन जावास्क्रिप्ट में इसे<code>a != b</code> लिखा जाता है।
 
-In this article we'll learn more about different types of comparisons, how JavaScript makes them, including important peculiarities. 
+इस लेख में हम विभिन्न प्रकार की तुलनाओं के बारे में अधिक जानेंगे, कि कैसे जावास्क्रिप्ट उन्हें महत्वपूर्ण विशिष्टता सहित बनाता है।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:12]: There should be a new line following this line. We try to maintain white space and blank lines. This makes it easier to merge any changes from the original tutorial.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 5]
```diff
@@ -1,214 +1,191 @@
...
+अंत में आपको "जावास्क्रिप्ट क्विर्क" से संबंधित मुद्दों से बचने के लिए एक अच्छा नुस्खा मिलेगा।
 
-At the end you'll find a good recipe to avoid "javascript quirks"-related issues.
+# बुलियन परिणाम है
+सभी तुलना ऑपरेटर एक बूलियन मान लौटाते हैं:
 
-## Boolean is the result
+- सत्य - का अर्थ है "हाँ", "सही" या "सत्य"।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:18]: true, false are javascript keywords hence we shouldn't be translating them.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 6]
```diff
@@ -1,214 +1,191 @@
...
+``` js run
+let result = 5 > 4; // तुलना का परिणाम बताएं
+alert( result ); // सच
 ```
 
-## String comparison
-
-To see whether a string is greater than another, JavaScript uses the so-called "dictionary" or "lexicographical" order.
+# स्ट्रिंग तुलना
+यह देखने के लिए कि क्या एक स्ट्रिंग दूसरे से बड़ी है, जावास्क्रिप्ट तथाकथित "शब्दकोश" या "लेक्सिकोग्राफिक" आदेश का उपयोग करता है।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:35]: We should use **लेक्सिकोग्राफिक** for **lexicographical** as it is easy to refer if required IMO.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 7]
```diff
@@ -1,214 +1,191 @@
...
-The algorithm to compare two strings is simple:
+दो स्ट्रिंग की तुलना करने के लिए एल्गोरिथ्म सरल है:
 
-1. Compare the first character of both strings.
-2. If the first character from the first string is greater (or less) than the other string's, then the first string is greater (or less) than the second. We're done.
-3. Otherwise, if both strings' first characters are the same, compare the second characters the same way.
-4. Repeat until the end of either string.
-5. If both strings end at the same length, then they are equal. Otherwise, the longer string is greater.
+1. दोनों स्ट्रिंगों के पहले चरित्र की तुलना करें।
+2. यदि पहली स्ट्रिंग से पहला वर्ण दूसरे स्ट्रिंग की तुलना में अधिक (या कम) है, तो पहली स्ट्रिंग दूसरी की तुलना में अधिक (या कम) है। हमारा काम हो गया।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:49]: Yes, character translated as **वर्ण** sounds more accurate to me. Lets update in other places if you agree.

File: 1-js/02-first-steps/09-comparison/article.md [hunk 8]
```diff
@@ -1,214 +1,191 @@
...
+3. अन्यथा, यदि दोनों तारों के पहले चरित्र समान हैं, तो दूसरे वर्णों की तुलना उसी तरह करें।
```
Comment (theBstar) [inline 1-js/02-first-steps/09-comparison/article.md:50]: string doesn't refer to **तारों** here. The translation tool must be confused. Someone was asking me why we don't use google translator and translate everything at one go. This exactly why we need human contributor. Machine intelligence in terms of Natual Language Processing is not there yet. Also, these translation tools are very generic.

**Review by theBstar — CHANGES_REQUESTED** [review]
Hey, first of all, huge appreciation for taking time to help a good cause. I went through the translation and found a couple of places where we can improve. Good test is to come back to the article after 30 minutes and see if every line is easy to understand. Translation doesn't need to be exact but it should convey the topic easily. Please resolve the thread and make any other required change. I will review the article post changes. Let me know if I can help in any way. :clap:

**Comment by theBstar** [comment]
@rjitsu I would love to merge this PR ASAP if we can make those small changes. Let me know if you are facing any difficulties.

**Comment by rjitsu** [comment]
Hi,
Sorry for the delayed PR, will make the changes over the weekend!

File: 1-js/02-first-steps/09-comparison/1-comparison-questions/solution.md (hunk 1 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/1-comparison-questions/solution.md:17]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 2 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:3]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 4 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:12]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 6 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:35]: I think I used लेक्सिकोग्राफिक for lexicographical only. Don't know if I'm missing something here

Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:35]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 7 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:49]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 8 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:50]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 5 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:18]: /done

File: 1-js/02-first-steps/09-comparison/article.md (hunk 3 above)
Comment (rjitsu) [reply 1-js/02-first-steps/09-comparison/article.md:7]: @theBstar which link? Can you help me with this, I have added all other changes that were requested

**Pull request closed without merging**